src/main/scala/beam/agentsim/scheduler/BeamAgentScheduler.scala

```diff
@@ -393,19 +419,21 @@ class BeamAgentScheduler(
         ) {
           if (nowInSeconds > 0 && nowInSeconds % 1800 == 0) {
             log.info(
-              "Hour " + nowInSeconds / 3600.0 + " completed. " + math.round(
+              "Hour " + nowInSeconds / 3600.0 + " completed. Longest delay " + longestNowUpdateDelay + " ms. "
+              + math.round(
                 10 * (Runtime.getRuntime.totalMemory() - Runtime.getRuntime
                   .freeMemory()) / Math
                   .pow(1000, 3)
               ) / 10.0 + "(GB)"
             )
+            longestNowUpdateDelay = 0
```

Comment:
Just because an hour completed doesn't mean the longest update should reset? It would still be valid that the longest delay is the longest delay...

Reply:
This longest delay is just a hint for the user about how long his simulation is delayed on some simulation time. I think that it's more interesting to see this delay for each hour comparing to the whole simulation.